lib/Sema/DerivedConformanceDifferentiable.cpp

```diff
@@ -39,14 +40,23 @@ getStoredPropertiesForDifferentiation(NominalTypeDecl *nominal, DeclContext *DC,
   auto &C = nominal->getASTContext();
   auto *diffableProto = C.getProtocol(KnownProtocolKind::Differentiable);
   for (auto *vd : nominal->getStoredProperties()) {
+    // Peer through property wrappers: use original wrapped properties instead.
+    if (auto *originalProperty = vd->getOriginalWrappedProperty()) {
+      // Skip property wrappers that do not define `wrappedValue.set`.
+      // `mutating func move(along:)` cannot be synthesized to update these
+      // properties.
+      auto *wrapperDecl =
+          vd->getInterfaceType()->getNominalOrBoundGenericNominal();
+      auto *wrappedValueDecl =
+          wrapperDecl->getPropertyWrapperTypeInfo().valueVar;
+      if (!wrappedValueDecl->getAccessor(AccessorKind::Set))
```

Comment:
I think you can perform this check on `originalProperty` directly - it won’t have a setter if the `wrappedValue` doesn’t have one.

Reply:
Done. Thanks for the review!